test.sh

```diff
@@ -1,20 +1,25 @@
 #!/bin/bash
+set -o errexit # Exit on errors
+
+LIBS="lib/* tests/*" # Library sources
 
 function test {
-  in=$1
-  out=bin/$(basename $in .cpp)
+  # Compile and run a test file
+  test_file=$1
+  binary=bin/$(basename $test_file .cpp)
 
-  echo Cleaning $out
-  rm -f $out
+  # Clean output
+  echo Cleaning $binary
+  rm -f $binary
 
-  echo Building $in
-  clang++ \
-    -g $in \
-    -o $out
+  # Build test
+  echo Building $test_file
+  clang++ -std=c++17 $test_file $LIBS
+  mv a.out $binary
```

Comment:
Not sure how to force clang to output directly to $binary
- clang cannot take the -o argument if given multiple files